src/platform/macos/system/surface.rs

```diff
@@ -461,16 +492,9 @@ impl<'a> Drop for SurfaceDataGuard<'a> {
     fn drop(&mut self) {
         unsafe {
             let mut seed = 0;
-            IOSurfaceUnlock(self.surface.io_surface.as_concrete_TypeRef(), 0, &mut seed);
-        }
-    }
-}
-
-impl Drop for NativeWidget {
-    #[inline]
-    fn drop(&mut self) {
-        unsafe {
-            let () = msg_send![self.view.0, release];
+            self.surface
+                .io_surface
+                .unlock(IOSurfaceLockOptions::empty(), &mut seed);
         }
     }
 }
```

Comment:
This kind of manual memory management is now unnecessary, the `Retained<NSView>` in `NativeWidget` will do it for you.